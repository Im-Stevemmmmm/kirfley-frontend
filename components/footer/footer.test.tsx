import { render } from "@testing-library/react";
import Footer from "components/footer/footer";

test("footer renders correctly", () => {
    const { container } = render(<Footer />);

    expect(container).toMatchSnapshot();
});

describe("about column", () => {
    it("is in the footer", () => {
        const { getByText } = render(<Footer />);
        const aboutColumn = getByText(/About/).parentElement;

        expect(aboutColumn).toBeInTheDocument;
    });

    it("has a company link", () => {
        const { getByText } = render(<Footer />);
        const aboutColumn = getByText(/About/).parentElement;

        expect(aboutColumn.children.item(1));
        expect(aboutColumn.children.item(1).textContent).toBe("Company");
    });

    it("has a business link", () => {
        const { getByText } = render(<Footer />);
        const aboutColumn = getByText(/About/).parentElement;

        expect(aboutColumn.children.item(2).textContent).toBe("Business");
    });
});

describe("support column", () => {
    it("is in the footer", () => {
        const { getByText } = render(<Footer />);
        const supportColumn = getByText(/Support/).parentElement;

        expect(supportColumn).toBeInTheDocument;
    });

    it("has a contact us link", () => {
        const { getByText } = render(<Footer />);
        const supportColumn = getByText(/Support/).parentElement;

        expect(supportColumn.children.item(1).textContent).toBe("Contact Us");
    });
});

describe("legal column", () => {
    it("is in the footer", () => {
        const { getByText } = render(<Footer />);
        const legalColumn = getByText(/Legal/).parentElement;

        expect(legalColumn).toBeInTheDocument;
    });

    it("has a privacy policy link", () => {
        const { getByText } = render(<Footer />);
        const legalColumn = getByText(/Legal/).parentElement;

        expect(legalColumn.children.item(1).textContent).toBe("Privacy Policy");
    });

    it("has a cookie policy link", () => {
        const { getByText } = render(<Footer />);
        const legalColumn = getByText(/Legal/).parentElement;

        expect(legalColumn.children.item(2).textContent).toBe("Cookie Policy");
    });

    it("has a copyright notice", () => {
        const { getByText } = render(<Footer />);
        const legalColumn = getByText(/Legal/).parentElement;

        expect(legalColumn.children.item(3).textContent).toBe(
            "\u00A9 Fern 2020"
        );
    });
});
