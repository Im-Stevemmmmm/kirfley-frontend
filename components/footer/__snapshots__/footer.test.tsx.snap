// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer renders correctly 1`] = `
<div>
  <footer
    id="footer"
  >
    <ul
      class="column"
    >
      <li
        class="column_title"
      >
        About
      </li>
      <li
        class="column_link"
        title="About the company behind Fern"
      >
        Company
      </li>
      <li
        class="column_link"
        title="About the company behind Fern"
      >
        Business
      </li>
    </ul>
    <ul
      class="column"
    >
      <li
        class="column_title"
      >
        Support
      </li>
      <li
        class="column_link"
        title="Get help with any problems"
      >
        Contact Us
      </li>
    </ul>
    <ul
      class="column"
    >
      <li
        class="column_title"
      >
        Legal
      </li>
      <li
        class="column_link"
        title="Our privacy policy"
      >
        Privacy Policy
      </li>
      <li
        class="column_link"
        title="Our cookie policy"
      >
        Cookie Policy
      </li>
      <li
        class="column_link column_link__copyright"
      >
        Â© Fern 
        2020
      </li>
    </ul>
  </footer>
</div>
`;
